builtin/providers/kubernetes/structures.go

```diff
@@ -198,3 +199,89 @@ func newStringSet(f schema.SchemaSetFunc, in []string) *schema.Set {
 	}
 	return schema.NewSet(f, out)
 }
+
+func schemaSetToStringArray(set *schema.Set) []string {
+	array := make([]string, 0, set.Len())
+	for _, elem := range set.List() {
+		e := elem.(string)
+		array = append(array, e)
+	}
+	return array
+}
+
+func flattenLocalObjectReferenceArray(in []api.LocalObjectReference) []interface{} {
+	att := make([]interface{}, len(in))
+	for i, v := range in {
+		m := map[string]interface{}{}
+		if v.Name != "" {
+			m["name"] = v.Name
+		}
+		att[i] = m
+	}
+	return att
+}
+
+func flattenPersistentVolumeClaimVolumeSource(in *api.PersistentVolumeClaimVolumeSource) []interface{} {
+	att := make(map[string]interface{})
+	if in.ClaimName != "" {
+		att["claim_name"] = in.ClaimName
+	}
+	if in.ReadOnly {
+		att["read_only"] = in.ReadOnly
+	}
+
+	return []interface{}{att}
+}
+
+func expandPersistentVolumeClaimVolumeSource(l []interface{}) *api.PersistentVolumeClaimVolumeSource {
+	if len(l) == 0 || l[0] == nil {
+		return &api.PersistentVolumeClaimVolumeSource{}
+	}
+	in := l[0].(map[string]interface{})
+	obj := &api.PersistentVolumeClaimVolumeSource{
+		ClaimName: in["claim_name"].(string),
+		ReadOnly:  in["read_only"].(bool),
+	}
+	return obj
+}
+
+func flattenVolumes(volumes []api.Volume) []interface{} {
+	att := make([]interface{}, len(volumes))
+	for i, v := range volumes {
+		obj := map[string]interface{}{}
+		if v.Name != "" {
+			obj["name"] = v.Name
+		}
+		if v.PersistentVolumeClaim != nil {
+			obj["persistent_volume_claim"] = flattenPersistentVolumeClaimVolumeSource(v.PersistentVolumeClaim)
+		}
+		//More values needed here
+		att[i] = obj
+	}
+	return att
+}
+
+func expandVolumes(volumes []interface{}) ([]api.Volume, error) {
+	if len(volumes) == 0 {
+		return []api.Volume{}, nil
+	}
+	vl := make([]api.Volume, len(volumes))
+	for i, c := range volumes {
+		v := c.(map[string]interface{})
```

Comment:
Just for clarity it would be nice to perhaps call it `m` (as "map") rather than `v`. That will free you up that variable name below where you can use it as "value".

While "value" is not great variable name, the map here is more significant because it is used in more than 2 lines - as opposed to 3-line conditionals where it matters less.